Ryujinx.HLE/HOS/Services/Hid/HidDevices/NpadDevices.cs

```diff
@@ -0,0 +1,334 @@
+                    defaultHeader.Type = ControllerType.JoyconLeft;
+                    defaultHeader.IsHalf = true;
+                    controller.DeviceType = DeviceType.JoyLeft;
+                    controller.SystemProperties |= NpadSystemProperties.SlSrButtonOriented |
+                                                   NpadSystemProperties.MinusButtonCapability;
+                    break;
+                case ControllerType.JoyconRight:
+                    defaultHeader.Type = ControllerType.JoyconRight;
+                    defaultHeader.IsHalf = true;
+                    controller.DeviceType = DeviceType.JoyRight;
+                    controller.SystemProperties |= NpadSystemProperties.SlSrButtonOriented |
+                                                   NpadSystemProperties.PlusButtonCapability;
+                    break;
+                case ControllerType.Pokeball:
+                    defaultHeader.Type = ControllerType.Pokeball;
+                    controller.DeviceType = DeviceType.Palma;
+                    break;
+            }
+
+            controller.Header = defaultHeader;
+
+            if (PrimaryController == PlayerIndex.Unknown)
+            {
+                PrimaryController = player;
+            }
+
+            _configuredNpads[(int)player].State = FilterState.Accepted;
+
+            _styleSetUpdateEvents[(int)player].ReadableEvent.Signal();
+
+            Logger.PrintInfo(LogClass.Hid, $"Connected ControllerType {type} to PlayerIndex {player}");
+        }
+
+        static NpadLayoutsIndex ControllerTypeToLayout(ControllerType controllerType)
+        => controllerType switch
+        {
+            ControllerType.ProController => NpadLayoutsIndex.ProController,
+            ControllerType.Handheld => NpadLayoutsIndex.Handheld,
+            ControllerType.JoyconPair => NpadLayoutsIndex.JoyDual,
+            ControllerType.JoyconLeft => NpadLayoutsIndex.JoyLeft,
+            ControllerType.JoyconRight => NpadLayoutsIndex.JoyRight,
+            ControllerType.Pokeball => NpadLayoutsIndex.Pokeball,
+            _ => NpadLayoutsIndex.SystemExternal
+        };
+
+        public void SetGamepadsInput(params GamepadInput[] states)
+        {
+            UpdateAllEntries();
+
+            for (int i = 0; i < states.Length; ++i)
+            {
+                SetGamepadState(states[i].PlayerId, states[i].Buttons, states[i].LStick, states[i].RStick);
+            }
+        }
+
+        void SetGamepadState(PlayerIndex player, ControllerKeys buttons,
+                    JoystickPosition leftJoystick, JoystickPosition rightJoystick)
+        {
+            if (player == PlayerIndex.Auto)
+            {
+                player = PrimaryController;
+            }
+            if (player == PlayerIndex.Unknown)
+            {
+                return;
+            }
+
+            NpadConfig p = _configuredNpads[(int)player];
+
+            if (p.State != FilterState.Accepted)
```

Comment:
```suggestion
            if (_configuredNpads[(int)player].State != FilterState.Accepted)
```
And remove the assign above since you don’t *p* elsewhere ?